### ISSUE TITLE
smaller than microsecond timedelta64 Series are not saved correctly with to_csv

### ISSUE THREAD

**Issue by cpcloud** [pandas-dev/pandas#6783]
```
In [24]: x = Series(np.around(randn(10) * 100).astype('timedelta64[ns]'))

In [25]: x
Out[25]:
0   -00:00:00.000000
1    00:00:00.000000
2   -00:00:00.000000
3   -00:00:00.000000
4    00:00:00.000000
5   -00:00:00.000000
6   -00:00:00.000000
7   -00:00:00.000000
8   -00:00:00.000000
9   -00:00:00.000000
dtype: timedelta64[ns]

In [26]: x.values
Out[26]: array([-259,   31,  -49,  -41,  141,  -14, -111, -117,  -20,  -35], dtype='timedelta64[ns]')

In [27]: buf = StringIO()

In [28]: x.to_csv(buf)

In [29]: print(buf.getvalue())
0,-00:00:00.000000
1,00:00:00.000000
2,-00:00:00.000000
3,-00:00:00.000000
4,00:00:00.000000
5,-00:00:00.000000
6,-00:00:00.000000
7,-00:00:00.000000
8,-00:00:00.000000
9,-00:00:00.000000
```

**Comment by mroeschke** [comment]
Looks like this is fixed on master. Could use a test

```
In [8]: print(buf.getvalue())
0,-1 days +23:59:59.999999925
1,-1 days +23:59:59.999999964
2,-1 days +23:59:59.999999957
3,-1 days +23:59:59.999999912
4,-1 days +23:59:59.999999996
5,0 days 00:00:00.000000094
6,0 days 00:00:00.000000130
7,-1 days +23:59:59.999999893
8,0 days 00:00:00.000000078
9,0 days 00:00:00.000000016
```

**Comment by baevpetr** [comment]
Hi, can I take it ?

**Comment by TomAugspurger** [comment]
@baevpetr I think this is already being fixed in #30554.
